File: src/main/java/com/kamikazejam/datastore/util/jackson/JacksonSpigotModule.java
package com.kamikazejam.datastore.util.jackson;

import com.fasterxml.jackson.core.Version;
import com.fasterxml.jackson.databind.module.SimpleModule;
import com.kamikazejam.datastore.util.jackson.deserialize.BlockDeserializer;
import com.kamikazejam.datastore.util.jackson.serialize.BlockSerializer;
import org.bukkit.block.Block;

public class JacksonSpigotModule extends SimpleModule {
    public JacksonSpigotModule() {
        super("JacksonSpigotModule", Version.unknownVersion());
        // Block
        addSerializer(Block.class, new BlockSerializer());
        addDeserializer(Block.class, new BlockDeserializer());
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/util/jackson/JacksonSpigotModule.kt
package com.kamikazejam.datastore.util.jackson

import com.fasterxml.jackson.core.Version
import com.fasterxml.jackson.databind.module.SimpleModule
import com.kamikazejam.datastore.util.jackson.deserialize.BlockDeserializer
import com.kamikazejam.datastore.util.jackson.serialize.BlockSerializer
import org.bukkit.block.Block

class JacksonSpigotModule : SimpleModule("JacksonSpigotModule", Version.unknownVersion()) {
    init {
        // Block
        addSerializer(Block::class.java, BlockSerializer())
        addDeserializer(Block::class.java, BlockDeserializer())
    }
}
